.github/workflows/build_and_test_cmake.yml

```diff
@@ -0,0 +1,47 @@
+name: Build and Test w/CMake
+permissions: read-all
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: seanmiddleditch/gha-setup-ninja@master
+
+      - name: Install prerequisites
+        run: |
+          sudo apt update
+          sudo apt install -y uuid-dev
+
+      - name: Cache LLVM artifact
+        id: cache-llvm
+        uses: actions/cache@v3
+        with:
+          path: |
+            ./externals/llvm-project
+          key: ${{ runner.os }}-norm-${{ hashFiles('externals/llvm-project/llvm/CMakeLists.txt') }}
+
+      - name: Build LLVM
+        if: steps.cache-llvm.outputs.cache-hit != 'true'
+        run: |
+          git submodule update --init --recursive
+          cd externals/llvm-project
+          mkdir build && cd build
+          cmake -G Ninja ../llvm -DLLVM_ENABLE_PROJECTS=mlir -DLLVM_BUILD_EXAMPLES=ON -DLLVM_ENABLE_ASSERTIONS=ON -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_RTTI=ON -DLLVM_TARGETS_TO_BUILD="host"
+          cmake --build . --target check-mlir
+
+      - name: Build and test mlir-tutorial
+        run: |
+          mkdir build && cd build
+          cmake -DLLVM_DIR=$PWD/../externals/llvm-project/build/lib/cmake/llvm -DMLIR_DIR=$PWD/../externals/llvm-project/build/lib/cmake/mlir ..
```

Comment:
nit: might be better to use `$github_workspace/externals/...` instead of the relative version

Reply:
indeed, will change that.